src/Lean/Server/FileWorker.lean

```diff
@@ -195,7 +195,11 @@ section Initialization
                                    srcPath.mapM realPathNormalized
       | _                       => throw <| IO.userError s!"unexpected output from `leanpkg print-paths`:\n{stdout}\nstderr:\n{stderr}"
     else
-      throw <| IO.userError s!"`leanpkg print-paths` failed:\n{stdout}\nstderr:\n{stderr}"
+      -- HACK(WN): `leanpkg` currently runs in the current directory, most likely the directory where the editor
```

Comment:
@Vtec234 Did you observe such a failure? In my tests, the server binary was always started in the workspace root.

Reply:
Yes, it happens on Windows. But you may also be able to observe it on Linux by adding multiple folders to a single workspace, at which point `leanpkg` for all folders will run in the first folder.

Reply:
At least in Emacs, I don't think we use multi-root servers because we didn't explicitly allow it. Should we? The benefit seems negligible.

Reply:
I often use multi-root workspaces so I would like them to work. We could either explicitly hande multi-root in our server or just launch a server per root. Unfortunately it seems like LSP prefers the former with `workspaceFolders`, meaning we cannot expect clients to launch multiple servers.

Reply:
https://github.com/microsoft/vscode/wiki/Adopting-Multi-Root-Workspace-APIs#single-language-server-or-server-per-folder doesn't make it sound like running multiple servers is deprecated or anything

Reply:
Ah, great!

Reply:
And for the sake of completeness, using multiple servers is important for development where we want to use a stage 0 server in `src/` and a stage 1 server in e.g. `tests/`

Reply:
The only issue is that this is client-side, so it would require special support from all clients. For example [here](https://github.com/microsoft/vscode-extension-samples/tree/fca30a52596381461f96548a55b50d345406e9bd/lsp-multi-server-sample) in VSCode. Also in that example nested folders do not have additional servers started (so one root in `lean4/` and another in `tests/` would only be one server), but I assume we can just change that.

Reply:
Yes, Emacs doesn't seem to have this subdirectory logic